File: Python/kyu6kata/find_the_parity_outlier.py
"""
You are given an array (which will have a length of at least 3, but could be very large) containing integers. The array is either entirely comprised of odd integers or entirely comprised of even integers except for a single integer N. Write a method that takes the array as an argument and returns this "outlier" N.

Examples
[2, 4, 0, 100, 4, 11, 2602, 36]
Should return: 11 (the only odd number)

[160, 3, 1719, 19, 11, 13, -21]
Should return: 160 (the only even number)
"""

def mine_find_outlier(integers):
    odd = 0
    even = 0
    for i in range(3):
        if integers[i] % 2 == 0:
            even += 1
        else:
            odd += 1
    if odd > even:
        for i in integers:
            if i % 2 == 0:
                return i
    else:
        for i in integers:
            if i % 2 != 0:
                return i
            
def best_find_outlier(int):
    odds = [x for x in int if x%2!=0]
    evens = [x for x in int if x%2==0]
    return odds[0] if len(odds)<len(evens) else evens[0]

File: Php/kyu6kata/findTheParityOutlier.php
<?php

function mineFind($integers) {
    $odds = [];
    $evens = [];
    foreach ($integers as $num)
    {
      if($num % 2 == 0)
      {
        array_push($evens, $num);
      } else {
        array_push($odds, $num);
      }
    }
    return (count($evens) > count($odds)) ? $odds[0] : $evens[0];
}

function bestFind($a) {
    $odds = [];
    $evens = [];
    foreach ($a as $n) {
      if ($n % 2) array_push($odds, $n);
      else array_push($evens, $n);
    }
    return count($evens) === 1 ? $evens[0] : $odds[0];
}

?>